objects/puzzles.rb

```diff
@@ -64,35 +64,44 @@ def group(xml)
   end
 
   def expose(xml, tickets)
+    exceptions = [] # TODO: Send exceptions to the user's email or admin
     seen = []
     Kernel.loop do
-      puzzles = xml.xpath(
-        '//puzzle[@alive="false" and issue
-        and issue != "unknown" and not(issue/@closed)' +
-        seen.map { |i| "and id != '#{i}'" }.join(' ') + ']'
-      )
-      break if puzzles.empty?
-      puzzle = puzzles[0]
-      puzzle.search('issue')[0]['closed'] = Time.now.iso8601 if tickets.close(puzzle)
-      save(xml)
+      begin
+        puzzles = xml.xpath(
+          '//puzzle[@alive="false" and issue
+          and issue != "unknown" and not(issue/@closed)' +
+          seen.map { |i| "and id != '#{i}'" }.join(' ') + ']'
+        )
+        break if puzzles.empty?
+        puzzle = puzzles[0]
+        puzzle.search('issue')[0]['closed'] = Time.now.iso8601 if tickets.close(puzzle)
+        save(xml)
+      rescue Exception => e
```

Comment:
@mbao01 I think we have a special decorator for catching exceptions here. If we don't, let's make it. Now you mix together two features: fetching puzzles and logging exceptions.